legacy/app/auth/authentication-interceptor.config.js

```diff
@@ -122,29 +52,16 @@ function AuthInterceptor($rootScope, $injector, $q, CONST, Session, _) {
                     $rootScope.$broadcast('event:unauthorized');
                     deferred.reject(rejection);
                 } else {
-                    // If we weren't logged in to start with
-                    // we probably just need to get a new token
-                    // (for some resources, of course,
-                    // this authorization level is not enough
-                    // and a 403 or 401 will be thrown
-                    // which results in showing the login page)
-
-                    // If this request was ignorable, ie ok to fail
-                    // just continue.
+                        // If this request was ignorable, ie ok to fail
+                        // just continue.
                     if (rejection.config.ignorable) {
                         deferred.reject(rejection);
                         return deferred.promise;
                     }
-
-                    concurrentGetClientCredsToken(rejection.config).then(
-                        function (config) {
-                            deferred.resolve($http(config));
-                        },
-                        deferred.reject
-                    );
+                    // Here we got the client-creds token previously. What to do now? Just reject the promise?
```

Comment:
> What to do now? Just reject the promise?
I think so. This function seems to be about error handling, so in that case it makes sense to me to exit it by propagating the error (to other error handlers, I assume)

In general, my opinion is that error handling routines should be as simple as possible. For that reason, I wouldn't say I like the original approach of getting a client token here... I'm glad we are getting rid of it.

Reply:
Agreed!